oz/__init__.py

```diff
@@ -131,7 +130,18 @@ def get_template_namespace(self):
     def initialize(self, **kwargs):
         self.trigger("initialize", **kwargs)
 
+    def set_secure_cookie(self, name, value, **kwargs):
+        if self.settings["use_secure_cookie"]:
+            if "secure" not in kwargs:
+                kwargs["secure"] = True
+            if "httponly" not in kwargs:
+                kwargs["httponly"] = True
+        super(RequestHandler, self).set_secure_cookie(name, value, **kwargs)
+
     def prepare(self):
+        if self.settings["use_hsts"]:
+            self.set_header("Strict-Transport-Security", "max-age=2592000; includeSubDomains")
```

Comment:
I don't think that increases vulnerability. The first request has already been sent over HTTP at that point. There's no way for us to stop that. It could have already been intercepted by a MITM. Also, the first visit to our web site is *very* likely to be from a logged-out user, so there would be no credentials.